src/@next/Upload/Upload.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { ChangeEvent, useEffect, useRef, useState } from 'react';
+import { Icon } from '../Icon';
+import {
+  StyledInvisibleInput,
+  StyledUploadContainer,
+  StyledTextContainer,
+  StyledIconContainer,
+  StyledUploadedImage,
+  StyledDeleteButtonContainer,
+  StyledDeleteIconContainer,
+  StyledErrorContainer,
+  StyledErrorIconContainer,
+  StyledErrorTextContainer,
+} from './UploadStyle';
+import { Spinner } from '../Spinner';
+import { Typography } from '../Typography';
+
+export interface UploadProps {
+  file: File | null;
+  setFile: React.Dispatch<any>;
+  loading?: boolean;
+  objectFit?: 'fill' | 'contain' | 'cover' | 'none' | 'scale-down';
+}
+
+export const Upload = React.forwardRef<HTMLInputElement, UploadProps>(
+  function Upload(
+    {
+      file,
+      setFile,
+      loading = false,
+      objectFit = 'cover',
+      ...props
+    }: UploadProps,
+    ref
+  ) {
+    const [attachmentUrl, setAttachmentUrl] = useState<string>('');
+    const [isLoading, setIsLoading] = useState<boolean>(loading);
+    const [error, setError] = useState<string>('');
+    const fileInputRef = useRef<HTMLInputElement>(null);
+
+    useEffect(() => {
+      const reader = new FileReader();
+      reader.onloadend = () => {
+        setAttachmentUrl(reader.result as string);
+        setIsLoading(false);
+      };
+      if (file) reader.readAsDataURL(file);
+    }, [file]);
+
+    const handleClick = () => {
+      if (fileInputRef.current) {
+        fileInputRef.current.click();
+      }
+    };
+
+    const MAX_FILE_SIZE = 5; // in MB
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+      setIsLoading(true);
+      const newFile = event.target.files?.[0];
+      if (newFile && !newFile.type.startsWith('image')) {
+        setError('Image required');
+        setFile(null);
+        setIsLoading(false);
+      } else if (newFile && newFile.size > MAX_FILE_SIZE * 1024 * 1024) {
+        setError('File too big');
+        setFile(null);
+        setIsLoading(false);
+      } else {
+        setError('');
+        setFile(newFile);
+      }
+    };
+
+    const handleDelete = () => {
+      setAttachmentUrl('');
+      setFile(null);
+    };
+
+    const unuploadedComponent = (
+      <StyledUploadContainer
+        onClick={handleClick}
+        data-type="initial"
+        data-error={!!error}
+        data-testid="upload-container"
+      >
+        <StyledIconContainer>
+          <Icon name="ri-add" />
+        </StyledIconContainer>
+        <StyledTextContainer>
+          <Typography as="span" variant="button">
+            Upload
```

Comment:
do keep in mind that we use i18n, it's better to expect a ReactNode for this and use "Upload" as default